src/modules/credentials/CredentialResponseCoordinator.ts

```diff
@@ -0,0 +1,188 @@
+ * Every function returns `true` if it should automate the flow and `false` if not
+ */
+@scoped(Lifecycle.ContainerScoped)
+export class CredentialResponseCoordinator {
+  private agentConfig: AgentConfig
+
+  public constructor(agentConfig: AgentConfig) {
+    this.agentConfig = agentConfig
+  }
+
+  /**
+   * Returns the credential auto accept config based on priority:
+   *	- The record config takes first priority
+   *	- Otherwise the agent config
+   *	- Otherwise {@link AutoAcceptCredential.Never} is returned
+   *
+   * @param recordConfig The auto accept config for the record
+   * @param agentConfig The auto accept config for the agent
+   * @returns the auto accept config
+   */
+  private static composeAutoAccept(
+    recordConfig: AutoAcceptCredential | undefined,
+    agentConfig: AutoAcceptCredential | undefined
+  ) {
+    return recordConfig ?? agentConfig ?? AutoAcceptCredential.Never
+  }
+
+  /**
+   * Checks whether it should automatically respond to a proposal
+   *
+   * @param credentialRecord The credentialRecord that contains the message(s) to respond to
+   * @param agentAutoAccept The configuration on the agent whether to auto accept
+   * @returns a message that will be send to the other agent
+   */
+  public async shoudlAutoRespondToProposal(credentialRecord: CredentialRecord) {
+    const autoAccept = CredentialResponseCoordinator.composeAutoAccept(
+      credentialRecord.autoAcceptCredential,
+      this.agentConfig.autoAcceptCredentials
+    )
+
+    if (autoAccept === AutoAcceptCredential.Always) {
+      return true
+    } else if (autoAccept === AutoAcceptCredential.ContentApproved) {
+      if (
+        credentialRecord.proposalMessage &&
+        credentialRecord.offerMessage &&
+        credentialRecord.proposalMessage.credentialProposal &&
+        credentialRecord.credentialAttributes
+      ) {
+        const proposalValues = CredentialUtils.convertAttributesToValues(
+          credentialRecord.proposalMessage.credentialProposal.attributes
+        )
+
+        const proposalCredentialDefinitionId = credentialRecord.proposalMessage.credentialDefinitionId
+
+        const defaultValues = CredentialUtils.convertAttributesToValues(credentialRecord.credentialAttributes)
+
+        const offerCredentialDefinitionId = credentialRecord.offerMessage.indyCredentialOffer?.cred_def_id
+
+        if (
+          CredentialUtils.checkValuesMatch(proposalValues, defaultValues) &&
+          proposalCredentialDefinitionId === offerCredentialDefinitionId
+        ) {
+          return true
+        }
+      }
+    }
+    return false
+  }
+
+  /**
+   * Checks whether it should automatically respond to an offer
+   *
+   * @param credentialRecord The credentialRecord that contains the message(s) to respond to
+   * @param agentAutoAccept The configuration on the agent whether to auto accept
+   * @returns a message that will be send to the other agent
```

Comment:
Good point recently made by @jakubkoci is that the `@param` and `@return` comments often don't add much value besides in the public API (which this isn't) and can be easily forgotten to be updated (as is the case here).